### PR TITLE
Enable AS Tabular Server mode

### PR DESCRIPTION
**Pull request by matt2005**
Added option to Install AS different server modes, this enables installation in Tabular mode.
https://msdn.microsoft.com/en-gb/library/hh231722.aspx

### CODE REVIEW

**Comment by matt2005** [comment]
AppVeyor error appears to be unrelated to pull request

File: DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1 [hunk 1]
```diff
@@ -121,6 +121,8 @@ function Get-TargetResource
             $ASServer.Connect("localhost\$InstanceName")
         }
         $ASCollation = ($ASServer.ServerProperties | Where-Object {$_.Name -eq "CollationName"}).Value
+        $ASSERVERMODE =  $(IF ($($ASServer.ServerMode) -eq 'Default') {'MULTIDIMENSIONAL'}
```
Comment (TravisEz13) [inline DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1:124]: Variable names should be camel cased.  [Style Guide - variable names should be camel cased](https://github.com/PowerShell/DscResources/blob/TravisEz13-patch-1/StyleGuidelines.md#variable-names-should-use-camelcase)

File: DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1 [hunk 2]
```diff
@@ -317,6 +320,9 @@ function Set-TargetResource
         [System.String]
         $ASCollation,
 
+        [System.String]
+        $ASSERVERMODE,
```
Comment (TravisEz13) [inline DSCResources/MSFT_xSQLServerSetup/MSFT_xSQLServerSetup.psm1:324]: Parameter names should be Pascal cased.

**Comment by TravisEz13** [comment]
It is required that you provide adequate test coverage for the code you change. [Contribution Guidelines -Adding test coverage for DSC resources](https://github.com/PowerShell/DscResources/blob/master/CONTRIBUTING.md#adding-test-coverage-for-dsc-resources)

**Comment by TravisEz13** [comment]
If there is no update of the pull request within 7 days, we will close the pull request.  After that, if you have an update, feel free to reopen the pull request or a new pull request.

**Comment by TravisEz13** [comment]
Closing because the pull request has not been updated or feedback given for 21 days.  Once you update the code, please feel free to reopen the pull request.  Thanks, Travis.

**Pull request closed without merging**